experiment/manual-trigger/manual-trigger.go

```diff
@@ -100,39 +102,45 @@ func sanityCheckFlags(o options) error {
 	return nil
 }
 
-func loadToken(path string) (string, error) {
-	raw, err := ioutil.ReadFile(path)
-	if err != nil {
-		return "", err
-	}
-	return strings.TrimSpace(string(raw)), nil
-}
-
 func main() {
 	o := flagOptions()
 	err := sanityCheckFlags(o)
 	if err != nil {
 		log.Fatal(err)
 	}
 
+	var tokens []string
+	tokens = append(tokens, o.githubTokenFile)
+
+	if o.jenkinsTokenFile != "" {
+		tokens = append(tokens, o.jenkinsTokenFile)
+	}
+
+	if o.jenkinsBearerTokenFile != "" {
+		tokens = append(tokens, o.jenkinsBearerTokenFile)
+	}
+
+	secretAgent := config.NewSecretAgent()
+	if err := secretAgent.Start(tokens); err != nil {
+		logrus.WithError(err).Fatal("Error starting secrets agent.")
+	}
+
+	getSecret := func(secretAgent *config.SecretAgent, secretPath string) func() []byte {
+		return func() []byte {
+			return secretAgent.GetSecret(secretPath)
+		}
+	}
+
 	// TODO(kargakis): dry this out
 	ac := jenkins.AuthConfig{}
 	if o.jenkinsTokenFile != "" {
-		token, err := loadToken(o.jenkinsTokenFile)
-		if err != nil {
-			log.Fatalf("cannot read file specified by --jenkins-token-file: %v", err)
-		}
 		ac.Basic = &jenkins.BasicAuthConfig{
 			User:  o.jenkinsUserName,
-			Token: token,
+			Token: getSecret(secretAgent, o.jenkinsTokenFile),
```

Comment:
Doesn't this mean that we load the Jenkins token just once? We should defer loading to when we need it or use a goroutine to keep a local copy in sync if the former approach is too resource intensive on every jenkins api call

Reply:
This is still outstanding?